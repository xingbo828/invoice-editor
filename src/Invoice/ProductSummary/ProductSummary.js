import React from 'react';
import { Link } from 'react-router-dom';

import Button from '../../Shared/Button';
import Icon from '../../Shared/Icon';
import {
  ProductSummaryList,
  ProductSummaryListHeading,
  ProductSummaryListItem
} from './styles';
const ProductSummary = ({ products, handleRemove }) => {
  const renderProductListItems = products => {
    return products.map(p => (
      <ProductSummaryListItem key={p.product.id}>
        <span><Link to="/catalogue">{p.product.title}</Link></span>
        <span>{p.quantity}</span>
        <span>${p.product.price}</span>
        <span>${p.quantity * p.product.price}</span>
        <span>
          <Button small danger onClick={handleRemove(p.product)}>
            <Icon size="lg" icon="trash" /><span style={{display: 'none'}}>Remove</span>
          </Button>
        </span>
      </ProductSummaryListItem>
    ));
  };
  return (
    <ProductSummaryList>
      <ProductSummaryListHeading>
        <span>Item</span>
        <span>Qty</span>
        <span>Price</span>
        <span>Total</span>
        <span>Action</span>
      </ProductSummaryListHeading>
      {renderProductListItems(products)}
    </ProductSummaryList>
  );
};

export default ProductSummary;
