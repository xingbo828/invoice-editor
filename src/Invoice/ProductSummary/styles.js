import styled from 'styled-components'

export const ProductSummaryList = styled.ul`
  padding: 0;
  margin: 2rem auto;
  max-width: 800px;
`

export const ProductSummaryListItem = styled.li`
  font-weight: ${props => props.theme.fontWeights.light};
  display: flex;
  justify-content: space-between;
  padding: ${props => props.theme.spaces.tight} 0;
  align-items: center;
  font-size: 1.25rem;
  border-bottom: 1px solid ${props => props.theme.colors.border};
  > * {
    flex: 1;
  }
`

export const ProductSummaryListHeading = styled(ProductSummaryListItem)`
  font-weight: ${props => props.theme.fontWeights.medium};
`