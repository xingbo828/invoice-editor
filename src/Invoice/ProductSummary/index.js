import { compose, withHandlers } from 'recompose'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { updateProduct } from '../../cartAction'
import { getProducts } from '../../cartReducer'


import ProductSummary from './ProductSummary'

const mapDispatchToProps = dispatch => bindActionCreators({
  updateProduct
}, dispatch)

const mapStateToProps = (state) => ({
  products: getProducts(state)
});

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),
  withHandlers({
    handleRemove: props => (product) => e => {
      props.updateProduct({
        product,
        quantity: 0
      })
    }
  })
)


export default enhance(ProductSummary)