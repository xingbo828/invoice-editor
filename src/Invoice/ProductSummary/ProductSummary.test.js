import React from 'react';
import { shallow } from 'enzyme';
import ProductSummary from './ProductSummary';

const props = {
  products: [{
    product: {
      title: 'a',
      price: 25,
      id: 'a'
    },
    quantity: 2
  }],
  handleRemove: () => {}
}
it('should match snapshot', () => {
  const wrapper = shallow(<ProductSummary {...props} />);
  expect(wrapper).toMatchSnapshot()
});