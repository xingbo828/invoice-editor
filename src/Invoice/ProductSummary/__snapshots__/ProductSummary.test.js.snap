// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductSummary
    handleRemove={[Function]}
    products={
      Array [
        Object {
          "product": Object {
            "id": "a",
            "price": 25,
            "title": "a",
          },
          "quantity": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <span>
            Item
          </span>
          <span>
            Qty
          </span>
          <span>
            Price
          </span>
          <span>
            Total
          </span>
          <span>
            Action
          </span>
        </ForwardRef>,
        Array [
          <ForwardRef>
            <span>
              <Link
                replace={false}
                to="/catalogue"
              >
                a
              </Link>
            </span>
            <span>
              2
            </span>
            <span>
              $
              25
            </span>
            <span>
              $
              50
            </span>
            <span>
              <Button
                danger={true}
                small={true}
                squared={false}
              >
                <Icon
                  icon="trash"
                  size="lg"
                  tag="i"
                />
                <span
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  Remove
                </span>
              </Button>
            </span>
          </ForwardRef>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <span>
              Item
            </span>,
            <span>
              Qty
            </span>,
            <span>
              Price
            </span>,
            <span>
              Total
            </span>,
            <span>
              Action
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Item",
            },
            "ref": null,
            "rendered": "Item",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Qty",
            },
            "ref": null,
            "rendered": "Qty",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Price",
            },
            "ref": null,
            "rendered": "Price",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Total",
            },
            "ref": null,
            "rendered": "Total",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Action",
            },
            "ref": null,
            "rendered": "Action",
            "type": "span",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              "
  font-weight: ",
              [Function],
              ";
  display: flex;
  justify-content: space-between;
  padding: ",
              [Function],
              " 0;
  align-items: center;
  font-size: 1.25rem;
  border-bottom: 1px solid ",
              [Function],
              ";
  > * {
    flex: 1;
  }
",
              "
  font-weight: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "Styled(styled.li)",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "li",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": "a",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <span>
              <Link
                replace={false}
                to="/catalogue"
              >
                a
              </Link>
            </span>,
            <span>
              2
            </span>,
            <span>
              $
              25
            </span>,
            <span>
              $
              50
            </span>,
            <span>
              <Button
                danger={true}
                small={true}
                squared={false}
              >
                <Icon
                  icon="trash"
                  size="lg"
                  tag="i"
                />
                <span
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  Remove
                </span>
              </Button>
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/catalogue"
              >
                a
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "a",
                "replace": false,
                "to": "/catalogue",
              },
              "ref": null,
              "rendered": "a",
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2,
            },
            "ref": null,
            "rendered": 2,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "$",
                25,
              ],
            },
            "ref": null,
            "rendered": Array [
              "$",
              25,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "$",
                50,
              ],
            },
            "ref": null,
            "rendered": Array [
              "$",
              50,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                danger={true}
                small={true}
                squared={false}
              >
                <Icon
                  icon="trash"
                  size="lg"
                  tag="i"
                />
                <span
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  Remove
                </span>
              </Button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="trash"
                    size="lg"
                    tag="i"
                  />,
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    Remove
                  </span>,
                ],
                "danger": true,
                "onClick": undefined,
                "small": true,
                "squared": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "trash",
                    "size": "lg",
                    "tag": "i",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Remove",
                    "style": Object {
                      "display": "none",
                    },
                  },
                  "ref": null,
                  "rendered": "Remove",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "rules": Array [
              "
  font-weight: ",
              [Function],
              ";
  display: flex;
  justify-content: space-between;
  padding: ",
              [Function],
              " 0;
  align-items: center;
  font-size: 1.25rem;
  border-bottom: 1px solid ",
              [Function],
              ";
  > * {
    flex: 1;
  }
",
            ],
          },
          "displayName": "styled.li",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "li",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  padding: 0;
  margin: 2rem auto;
  max-width: 800px;
",
        ],
      },
      "displayName": "styled.ul",
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "ul",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <span>
              Item
            </span>
            <span>
              Qty
            </span>
            <span>
              Price
            </span>
            <span>
              Total
            </span>
            <span>
              Action
            </span>
          </ForwardRef>,
          Array [
            <ForwardRef>
              <span>
                <Link
                  replace={false}
                  to="/catalogue"
                >
                  a
                </Link>
              </span>
              <span>
                2
              </span>
              <span>
                $
                25
              </span>
              <span>
                $
                50
              </span>
              <span>
                <Button
                  danger={true}
                  small={true}
                  squared={false}
                >
                  <Icon
                    icon="trash"
                    size="lg"
                    tag="i"
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    Remove
                  </span>
                </Button>
              </span>
            </ForwardRef>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                Item
              </span>,
              <span>
                Qty
              </span>,
              <span>
                Price
              </span>,
              <span>
                Total
              </span>,
              <span>
                Action
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item",
              },
              "ref": null,
              "rendered": "Item",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Qty",
              },
              "ref": null,
              "rendered": "Qty",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Price",
              },
              "ref": null,
              "rendered": "Price",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total",
              },
              "ref": null,
              "rendered": "Total",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Action",
              },
              "ref": null,
              "rendered": "Action",
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                "
  font-weight: ",
                [Function],
                ";
  display: flex;
  justify-content: space-between;
  padding: ",
                [Function],
                " 0;
  align-items: center;
  font-size: 1.25rem;
  border-bottom: 1px solid ",
                [Function],
                ";
  > * {
    flex: 1;
  }
",
                "
  font-weight: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "Styled(styled.li)",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "li",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                <Link
                  replace={false}
                  to="/catalogue"
                >
                  a
                </Link>
              </span>,
              <span>
                2
              </span>,
              <span>
                $
                25
              </span>,
              <span>
                $
                50
              </span>,
              <span>
                <Button
                  danger={true}
                  small={true}
                  squared={false}
                >
                  <Icon
                    icon="trash"
                    size="lg"
                    tag="i"
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    Remove
                  </span>
                </Button>
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/catalogue"
                >
                  a
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "a",
                  "replace": false,
                  "to": "/catalogue",
                },
                "ref": null,
                "rendered": "a",
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2,
              },
              "ref": null,
              "rendered": 2,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "$",
                  25,
                ],
              },
              "ref": null,
              "rendered": Array [
                "$",
                25,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "$",
                  50,
                ],
              },
              "ref": null,
              "rendered": Array [
                "$",
                50,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  danger={true}
                  small={true}
                  squared={false}
                >
                  <Icon
                    icon="trash"
                    size="lg"
                    tag="i"
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    Remove
                  </span>
                </Button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Icon
                      icon="trash"
                      size="lg"
                      tag="i"
                    />,
                    <span
                      style={
                        Object {
                          "display": "none",
                        }
                      }
                    >
                      Remove
                    </span>,
                  ],
                  "danger": true,
                  "onClick": undefined,
                  "small": true,
                  "squared": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "icon": "trash",
                      "size": "lg",
                      "tag": "i",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Remove",
                      "style": Object {
                        "display": "none",
                      },
                    },
                    "ref": null,
                    "rendered": "Remove",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "rules": Array [
                "
  font-weight: ",
                [Function],
                ";
  display: flex;
  justify-content: space-between;
  padding: ",
                [Function],
                " 0;
  align-items: center;
  font-size: 1.25rem;
  border-bottom: 1px solid ",
                [Function],
                ";
  > * {
    flex: 1;
  }
",
              ],
            },
            "displayName": "styled.li",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "li",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  padding: 0;
  margin: 2rem auto;
  max-width: 800px;
",
          ],
        },
        "displayName": "styled.ul",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "ul",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
